server/routes/return-of-service.js

```diff
@@ -98,29 +118,88 @@ router.get(
     keycloak.allowRolesMiddleware('ministry_of_health', 'health_authority', 'employer')
   ),
   asyncMiddleware(async (req, res) => {
+    const actionName = 'ros-status-get';
     const { participantId } = req.params;
     const { email, user_id: userId, sub: localUserId } = req.user;
     const user = userId || localUserId;
     if (!(email && user)) {
       return res.status(401).send('Unauthorized user');
     }
-    // Validate Participant
-    const [participant] = await getParticipantByID({ id: participantId });
+    const participant = await validateParticipant(participantId, actionName);
     if (!participant) {
-      logger.error({
-        action: 'ros-status-create',
-        message: `Participant ${participantId} not found`,
-      });
       return res.status(404).send('Participant not found');
     }
     const resp = await getReturnOfServiceStatuses({ participantId });
     logger.info({
-      action: 'ros-status-get',
+      action: actionName,
       performed_by: user,
       ids: resp.map((r) => r.id),
     });
     return res.status(200).json(resp);
   })
 );
 
+// Update return of service fields
+router.patch(
+  '/participant/:participantId',
+  applyMiddleware(keycloak.allowRolesMiddleware('ministry_of_health')),
+  asyncMiddleware(async (req, res) => {
+    const actionName = 'ros-status-update';
+    const { participantId } = req.params;
+    const { email, user_id: userId, sub: localUserId } = req.user;
+    const user = userId || localUserId;
+    if (!(email && user)) {
+      return res.status(401).send('Unauthorized user');
+    }
+    const participant = await validateParticipant(participantId, actionName);
+    if (!participant) {
+      return res.status(404).send('Participant not found');
+    }
+    const { newValue, fieldType } = req.body;
```

Comment:
I don't think I'm a fan of this pattern of passing in `fieldType`. Generally, I think a patch would be optional properties of an entity, so in this case you would do something like:
```
const { siteId, date, startDate } = req.body;
if (siteId)...
else if (date)...
else if (startDate)...
```

That said, I do agree with Arran in the comment on the three functions